### ISSUE TITLE
Serializing nested structures

### ISSUE THREAD

**Issue by timgabets** [RReverser/serde-xml-rs#126]
I've got a problem with serializing nested structures (the struct inside of another struct): 

```rust
// [dependencies]
// serde = { version = "1.0", features = ["derive"] }
// serde-xml-rs = "0.4.0"

use serde::Serialize;
use serde_xml_rs::to_string;

#[derive(Serialize, Debug)]
struct Header {
    message_id: i64,
    system_id: String,
}

#[derive(Serialize, Debug)]
struct Request {
    #[serde(flatten)]
    header: Header
}

impl Request {
	pub fn serialize(&self) -> String {
        to_string(&self).unwrap()
    }
}

fn main() {
    let req = Request{header: Header{message_id: 1234, system_id: String::from("IDDQD")}};

    println!("{:?}", req.serialize());
}
```

The problem is that after serialization there's two header tags (but I'd like to get one):
```xml
<Request><header><Header><message_id>1234</message_id><system_id>IDDQD</system_id></Header></header></Request>
```
Is there any ideas on how could I get something like this?
```xml
<Request><Header><message_id>1234</message_id><system_id>IDDQD</system_id></Header></Request>
```

P.S. Sorry, I'm not quite sure whether it's a serde-related bug/feauture or it's serde-xml-rs specific

**Comment by mmatoszko** [comment]
I'm also experiencing this.
@RReverser @punkstarman If this is something that could use some improvement, I would be happy to work on it :smiley:

**Comment by jmagnusson** [comment]
FYI I'm getting this result:
```xml
<message_id>1234</message_id><system_id>IDDQD</system_id>
```
with the following dependencies:
```
serde = { version = "1.0.126", features = ["derive"] }
serde-xml-rs = "0.4.1"
```

**Comment by Corasonn** [comment]
I also have this issue. 

serde = { version = "1.0.18", features = ["derive"] }
serde-xml-rs = "0.5.1"

Here is a solution/workaround:
https://stackoverflow.com/questions/70124048/how-to-create-xml-from-struct-in-rust

**Comment by punkstarman** [comment]
I am working to rewrite the serializer entirely. Just waiting to get some time. Hopefully, before the year is over.

**Comment by punkstarman** [comment]
Solved by #173

**Comment by zmrow** [comment]
Just running into this same issue now!  Any chance of a release that includes the fix? :) :)
